qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -0,0 +1,265 @@
+                    "ZX plain": False,
+                    "Theta": new_theta
+                }
+        else:
+            new_qubit_state = qubit_state
+        # new_qubit_state = qubit_state
+        return new_qubit_state
+
+    def _rx_gate(self, qubit_state: dict, angle: float) -> dict:
+        """
+        Apply Rx gate.
+        Args:
+            qubit_state(dict): The state of the qubit before operating the gate.
+            angle(float): The angle of the rotation.
+
+        Returns:
+                dict: The state of the qubit after operating the gate.
+        """
+        if qubit_state["XY plain"]:
+            if isclose(angle, np.pi):
+                new_theta = - qubit_state["Theta"]
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": True,
+                    "ZX plain": False,
+                    "Theta": new_theta,
+                }
+            elif isclose(angle, np.pi/2):
+                new_theta = angle - qubit_state["Theta"]
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": False,
+                    "ZX plain": True,
+                    "Theta": new_theta,
+                }
+            else:
+                print("Error - This angle isn't supported. We only support multipication of pi/2")
+        else:
+            if isclose(angle, np.pi/2):
+                new_theta = qubit_state["Theta"] + 3 * np.pi/2  # its theta -pi/2 but we added 2*pi
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": True,
+                    "ZX plain": False,
+                    "Theta": new_theta,
+                }
+            elif isclose(angle, np.pi):
+                new_theta = qubit_state["Theta"] + np.pi
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": False,
+                    "ZX plain": True,
+                    "Theta": new_theta,
+                }
+            else:
+                print("Error - This angle isn't supported. We only support multiplication of pi/2")
+        return new_qubit_state
+
+
+    def _measurement_gate(self, qubit_state: dict) -> int:
+        """
+        implementing measurement on qubit with read-out error.
+        Args:
+            qubit_state(dict): The state of the qubit at the end of the circuit.
+
+        Returns:
+                int: The result of the measurement after applying read-out error.
+        """
+        if qubit_state["XY plain"]:
+            meas_res = self._rng.random() < 0.5
+        else:
+            z_projection = np.cos(qubit_state["Theta"])
+            probability = abs(z_projection) ** 2
+            if self._rng.random() > probability:
+                meas_res = self._rng.random() < 0.5
+            else:
+                meas_res = z_projection < 0
```

Comment:
I think it'd be easier to calculate the statevector (the coefficients of |0> and |1>) and then perform one randomization.

Reply:
While writing the backend, I thought to not write it with statevectors. I think unless I change all the backend to be with state vectors, it will be unclear why suddenly I started to use statevectors and not stick to the end with the current representation.

Reply:
Ok. Then please add a comment that explains lines 192-197

Reply:
Added comments with explanation for the measurement process.